compiler/src/iree/compiler/Dialect/Flow/Conversion/MeshToFlow/test/channel_creation.mlir

```diff
@@ -0,0 +1,132 @@
+    // CHECK: %[[AXIS_0_SIZE:.*]] = arith.constant 3 : index
+    // CHECK: %[[DEFAULT_CHANNEL:.*]] = flow.channel.default : !flow.channel
+    // CHECK: %[[CHANNEL_RANK:.*]] = flow.channel.rank %[[DEFAULT_CHANNEL]] : index
+    // CHECK: %[[DEVICE_MULTI_IDX:.*]]:4 = affine.delinearize_index %[[CHANNEL_RANK]] into
+    // CHECK-SAME: (%[[AXIS_0_SIZE]], %[[AXIS_1_SIZE]], %[[AXIS_2_SIZE]], %[[AXIS_3_SIZE]]) : index, index, index, index
+    // CHECK: %[[IN_GROUP_IDX:.*]] = affine.apply
+    // CHECK-SAME: #map()[%[[DEVICE_MULTI_IDX]]#2, %[[DEVICE_MULTI_IDX]]#1, %[[AXIS_2_SIZE]], %[[AXIS_1_SIZE]]]
+    // CHECK: %[[GROUP_IDX:.*]] = affine.apply
+    // CHECK-SAME: #map()[%[[DEVICE_MULTI_IDX]]#0, %[[DEVICE_MULTI_IDX]]#3, %[[AXIS_0_SIZE]], %[[AXIS_3_SIZE]]]
+    // CHECK: %[[CHANNEL:.*]] = flow.channel.split
+    // CHECK-SAME: %[[DEFAULT_CHANNEL]], %[[GROUP_IDX]], %[[IN_GROUP_IDX]] : !flow.channel -> !flow.channel
+    // CHECK: util.global.store %[[CHANNEL]], @_mesh_mesh_4d_axes_2_1 : !flow.channel
+  mesh.cluster @mesh_4d(shape = 3x4x5x6)
+
+  func.func @f(%input : tensor<1xi8>) -> tensor<1xi8> {
+    %out = mesh.all_reduce %input on @mesh_4d mesh_axes = [2, 1] : tensor<1xi8> -> tensor<1xi8>
+    return %out : tensor<1xi8>
+  }
+}
+
+// -----
+
+// CHECK-LABEL: module @multiple_different_channels
+module @multiple_different_channels {
+
+  // CHECK-DAG: util.global public @_mesh_mesh_2d_axes_0 : !flow.channel
+  // CHECK-DAG: util.global public @_mesh_mesh_2d_axes_1 : !flow.channel
+  mesh.cluster @mesh_2d(shape = 3x4)
+
+  func.func @f(%input : tensor<1xi8>) -> (tensor<1xi8>, tensor<1xi8>) {
+    %out0 = mesh.all_reduce %input on @mesh_2d mesh_axes = [0] : tensor<1xi8> -> tensor<1xi8>
+    %out1 = mesh.all_reduce %input on @mesh_2d mesh_axes = [1] : tensor<1xi8> -> tensor<1xi8>
+    return %out0, %out1 : tensor<1xi8>, tensor<1xi8>
+  }
+}
+
+// -----
+
+// CHECK-LABEL: module @same_channel_used_multiple_times
+module @same_channel_used_multiple_times {
+
+  // CHECK: util.global public @_mesh_mesh_2d_axes_0 : !flow.channel
+  mesh.cluster @mesh_2d(shape = 3x4)
+
+  func.func @f(%input0 : tensor<1xi8>, %input1 : tensor<1xi8>) -> (tensor<1xi8>, tensor<1xi8>) {
+    %out0 = mesh.all_reduce %input0 on @mesh_2d mesh_axes = [0] : tensor<1xi8> -> tensor<1xi8>
+    %out1 = mesh.all_reduce %input1 on @mesh_2d mesh_axes = [0] : tensor<1xi8> -> tensor<1xi8>
+    return %out0, %out1 : tensor<1xi8>, tensor<1xi8>
+  }
+}
+
+// -----
+
+// CHECK-LABEL: module @multiple_meshes
+module @multiple_meshes {
+
+  // CHECK: util.global public @_mesh_mesh1_axes_0 : !flow.channel
+  // CHECK: util.initializer {
+    // CHECK: %[[DEFAULT_CHANNEL:.*]] = flow.channel.default "mesh1" : !flow.channel
+    // CHECK: %[[CHANNEL_RANK:.*]] = flow.channel.rank %[[DEFAULT_CHANNEL]] : index
+    // CHECK: %[[CHANNEL:.*]] = flow.channel.split
+    // CHECK-SAME: %[[DEFAULT_CHANNEL]], %{{.*}}, %{{.*}} : !flow.channel -> !flow.channel
```

Comment:
you probably want to check the arguments here (given that you're capturing CHANNEL_RANK but not using it)

Reply:
I removed the unused capturings. The full set of arguments of `flow.channel.split` is tested in another test. Here the concern is that `flow.channel.default` uses a name.